pkg/mimirtool/client/backfill.go

```diff
@@ -100,11 +101,20 @@ func (c *MimirClient) backfillBlock(ctx context.Context, blockDir string, logctx
 		}
 	}
 
-	resp, err = c.doRequest(ctx, path.Join(endpointPrefix, url.PathEscape(blockID), finishBlockUpload), http.MethodPost, nil, -1)
-	if err != nil {
-		return errors.Wrap(err, "request to finish block upload failed")
+	for {
+		resp, err = c.doRequest(ctx, path.Join(endpointPrefix, url.PathEscape(blockID), finishBlockUpload), http.MethodPost, nil, -1)
+		if err != nil {
+			if strings.Contains(err.Error(), "429 Too Many Requests") {
+				// wait and try again
+				logctx.WithField("error", err).Warning("will sleep and try again")
```

Comment:
We can give user some context, eg. `failed to finish block upload and start block validation due to too many requests, will sleep and try again`. WDYT?

Reply:
I see that there is more context given in the log already (full response body), and my suggested message wouldn't fit into the format.